pkg/v17/chartconfig/chartconfig.go

```diff
@@ -105,6 +105,9 @@ func filterChartOperatorAnnotations(cr *v1alpha1.ChartConfig) map[string]string
 	annotations := map[string]string{}
 
 	for k, v := range cr.Annotations {
+		if k == annotation.CordonReason || k == annotation.CordonUntilDate {
+			continue
+		}
```

Comment:
When cluster-operator found out there are other annotations value has prefix value with `chart-operator.giantswarm.io/` from current CRs, it just replace whole annotations as same as  `desiredChartConfig`. 
By introducing this PR, we can opt to not comparing cordon related annotation values.

Reply:
OK cool it makes sense not to compare on those annotations. But my worry is that we update the chartconfig CR rather than patching it.

https://github.com/giantswarm/cluster-operator/blob/2e83331d2350247d717e170ecc6c465181538cb6/pkg/v18/chartconfig/update.go#L22

So if the annotations are not in the desired state won't they be removed?

Reply:
If we have other chartconfig values had been changed, cordon related annotations would be removed. But I hardly expect that the situation would be happening.

Reply:
I moved it to WIP version atm, :) c4d3e92